File: store_credit/store_credit.go
package main

import f "fmt"

type Item struct {
	value int
	index int
}

func solve(credit int, items []Item) (Item, Item) {
	for i := 1; i < len(items); i++ {
		if (items[0].value + items[i].value == credit) {
			return items[0], items[i]
		}
	}
	return solve(credit, items[1:])
}

func main() {
	var n, credit, count, value int
	f.Scanf("%d", &n)
	for i := 1; i <= n; i++ {
		f.Scanf("%d", &credit)
		f.Scanf("%d", &count)
		items := make([]Item, count)
		for t := 0; t < count; t++ {
			f.Scanf("%d", &value)
			items[t] = Item{value, t + 1}
		}
		r1, r2 := solve(credit, items[:])
		i1, i2 :=  r1.index, r2.index
		if (i1 > i2) {
			i1, i2 = i2, i1
		}
		f.Printf("Case #%d: %d %d\n", i, i1, i2)
	}
}

File: store_credit/store_credit.cpp
#include <vector>
#include <algorithm>
#include <ext/functional>
#include <iostream>

using namespace std;
using namespace __gnu_cxx;

typedef pair<int, int> Item;
typedef vector<Item>::iterator vit;

pair<Item, Item> solve(int credit, vit pos, vit end) {
	Item head = *pos++;
	vit p = find_if(pos, end, compose1(
		bind2nd(equal_to<int>(), credit - head.first), 
		select1st<Item>()
	));
	if (p != end) {
		return pair<Item, Item>(head, *p);
	} 
	return solve(credit, pos, end);
}

int main() {
	int N, credit, count, value;
	cin >> N;
	for (int n = 1; n <= N; n++) {
		cin >> credit >> count;
		vector<Item> items;
		for (int i = 1; i <= count; i++) {
			cin >> value;
			if (value < credit) {
				items.push_back(Item(value, i));				
			}
		}
		pair<Item, Item> r = solve(credit,  items.begin(), items.end());
		int values[] = {r.first.second, r.second.second};
		sort(values, &values[2]);
		cout << "Case #" << n << ": " << values[0] << " " << values[1] << endl;
	}
	return 0;
}